sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/service/schema/common/SchemaManager.java

```diff
@@ -278,6 +359,59 @@ private Uni<Schema.CqlTable> getTableInternal(
             });
   }
 
+  // returns all keyspace names
+  private Multi<String> getKeyspaceNames(StargateBridge bridge) {
+    // repeated multi to achieve paging
+    return Multi.createBy()
+        .repeating()
+
+        // start with the optimistic state
+        .uni(
+            () -> new AtomicReference<>(new Paging(true, null)),
+            state -> {
+              // if we don't have more, short circuit
+              Paging paging = state.get();
+              if (!paging.hasMore()) {
+                return Uni.createFrom().item(Collections.<String>emptyList());
+              }
+
+              // otherwise execute the query
+              QueryOuterClass.Query.Builder queryBuilder =
+                  QueryOuterClass.Query.newBuilder()
+                      .setCql("SELECT keyspace_name FROM system_schema.keyspaces");
+              if (null != paging.pageState()) {
+                QueryOuterClass.QueryParameters.Builder params =
```

Comment:
should we set a bigger page size here, like `100` to actually avoid double trips in 99.999%?

Reply:
Done.